### PR TITLE
RUB-27: add React, add basic Template

### PR DESCRIPTION
**Pull request by vlodkopits**
# REVIEWERS

- [x] @UAVova 
- [x] @mayar2610 

# DESCRIPTION

**Add React(react-rails), add gem react-rails-img to render img, add basic Tamplate and Styles.**

# TEST COVERAGE

**no test.**

# EXPERT REVIEW

- [x] @rokor
- [ ] @okohuch

### CODE REVIEW

File: app/assets/stylesheets/main.scss [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    margin-bottom: 20px;
+}
+
+/* --- header --- */
+.navbar{
+	background-color: $burgundy;
+
+	li{
+		:hover{
+			color: $burgundy;
```
Comment (okohuch) [inline app/assets/stylesheets/main.scss]: Change tab to 2 spaces and fix alignment in this file.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -4,11 +4,21 @@
...
 
+    <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/bootstrap/3.0.3/css/bootstrap.css">
+    <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/bootstrap/3.0.3/css/bootstrap-theme.css">
     <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
 
   <body>
-    <%= yield %>
+  	<%= react_component("Navbar") %>
```
Comment (okohuch) [inline app/views/layouts/application.html.erb]: alignment should be 2 spaces

**Review by mayar2610 — APPROVED** [review]

**Review by intsrc — APPROVED** [review]

File: app/assets/javascripts/components/navbar.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    return (
+      <div>
+        <nav className="navbar">
+          <div className="container-fluid">
+            
+            <div className="navbar-header">
+              <button type="button" className="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar-collapse" aria-expanded="false">
+                <span className="sr-only">Toggle navigation</span>
+                <span className="icon-bar"></span>
```
Comment (rokor) [inline app/assets/javascripts/components/navbar.jsx]: duplication happening with span.

File: app/assets/javascripts/components/navbar.jsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+              </button>
+              <a className="navbar-brand" href={ this.state.urlRoot }>
+                <img src={ this.state.urlLogo } alt="Fight Club" />
+              </a>
+            </div>
+
+            <div className="collapse navbar-collapse" id="navbar-collapse">
+              <ul className="nav navbar-nav pull-right">
+                <li><a href={ this.state.urlSignin } >Sign In</a></li>
```
Comment (rokor) [inline app/assets/javascripts/components/navbar.jsx]: up to line 41 should be dynamically generated.

File: app/assets/javascripts/components/navbar.jsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
+class Navbar extends React.Component {
+  constructor() {
+   super();
+   this.state = {
+      urlRoot: "/",
```
Comment (rokor) [inline app/assets/javascripts/components/navbar.jsx]: should be moved to const in separate folder and file

File: app/assets/javascripts/components/footer.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+class Footer extends React.Component {
+  constructor() {
+   super();
+   this.state = {
+
```
Comment (rokor) [inline app/assets/javascripts/components/footer.jsx]: empty state not the best idea.

**Review by rokor — CHANGES_REQUESTED** [review]

File: app/assets/javascripts/components/navbar.jsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
+class Navbar extends React.Component {
+  constructor() {
+   super();
+  }
+
+  generate_tags(Tag, fields) {
+    const tags = fields.map((name, index) => {
+      return <Tag key={index} className={name}></Tag>;
```
Comment (rokor) [inline app/assets/javascripts/components/navbar.jsx]: () should follow after return when we generate elements. For example `return (<Tag key={index} className={name}></Tag>);`

File: app/assets/javascripts/components/navbar.jsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    });
+
+    return tags;
+  }
+
+  generate_links(links_list) {
+    const links = links_list.map((value, index) => {
+      return  <li key={index}>
```
Comment (rokor) [inline app/assets/javascripts/components/navbar.jsx]: same issue as above

**Review by rokor — APPROVED** [review]

**Pull request merged**